packages/vats/src/core/utils.js

```diff
@@ -212,19 +212,27 @@ const noop = harden(() => {});
 export const makePromiseSpaceForNameHub = (nameAdmin, log = noop) => {
   const logHooks = makeLogHooks(log);
 
+  const readonlyP = E(nameAdmin).readonly();
+
   /** @type {PromiseSpaceOf<any>} */
   const space = makePromiseSpace({
     hooks: harden({
       ...logHooks,
       onAddKey: name => {
-        void E(nameAdmin).reserve(name);
+        void E.sendOnly(nameAdmin).reserve(name);
         logHooks.onAddKey(name);
       },
       onResolve: (name, valueP) => {
-        void E.when(valueP, value => E(nameAdmin).update(name, value));
+        void E.when(valueP, value => E.sendOnly(nameAdmin).update(name, value));
       },
       onReset: name => {
-        void E(nameAdmin).delete(name);
+        void E(readonlyP)
+          .has(name)
+          .then(present => {
+            if (present) {
+              void E.sendOnly(nameAdmin).delete(name);
```

Comment:
seems worth a unit test in [test-name-hub-published.js](https://github.com/Agoric/agoric-sdk/blob/feefe9bab3ed25d7fa8700d63968228c46509da2/packages/vats/test/test-name-hub-published.js)

Reply:
but again... how do we deploy this change to mainnet?

Maybe the answer is "we don't" but that merits something explicit in upgrade considerations.

Reply:
This change is outdated since I simplified the `utils.js` change to just one line.